### PR TITLE
Search auth based on dashboard_acl

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
to https://github.com/grafana/grafana/pull/45430

### CODE REVIEW

File: pkg/services/searchV2/service.go [hunk 1]
```diff
@@ -41,20 +41,33 @@ type dashMeta struct {
...
 	// Load and parse all dashboards for given orgId
 	dash, err := loadDashboards(ctx, orgId, s.sql)
 	if err != nil {
 		rsp.Error = err
 		return rsp
 	}
 
-	// TODO!!! get user from context?
-	dash, err = s.applyAuthFilter(nil, dash)
+	getSignedInUserQuery := &models.GetSignedInUserQuery{
```
Comment (FZambia) [inline pkg/services/searchV2/service.go:51]: Smart :) Maybe leave a comment to not forget to do sth with this in the future?

File: pkg/services/searchV2/auth.go [hunk 2]
```diff
@@ -19,14 +20,62 @@ type simpleSQLAuthService struct {
...
+	if err != nil {
+		return nil, err
+	}
+
+	var uids []string
+	for i := 0; i < len(rows); i++ {
+		uids = append(uids, rows[i].UID)
+	}
+
+	return uidFilter{uids: uids}.filter, err
```
Comment (FZambia) [inline pkg/services/searchV2/auth.go:64]:
```suggestion
	return &uidFilter{uids: uids}.filter, err
```
May resolve lint issue - but better to re-check

**Review by FZambia — COMMENTED** [review]
Looks nice, just couple of comments

File: pkg/services/searchV2/service.go (hunk 1 above)
Comment (ArturWierzbicki) [reply pkg/services/searchV2/service.go:52]: yup, readded!

File: pkg/services/searchV2/auth.go (hunk 2 above)
Comment (ArturWierzbicki) [reply pkg/services/searchV2/auth.go]: transformed `resourceFilter` into an interface to fix this

Comment (FZambia) [reply pkg/services/searchV2/auth.go]: Sorry - this won't work, need to define a separate variable. Maybe if ResourceFilter was an interface things were more readable.

File: pkg/services/searchV2/service.go [hunk 3]
```diff
@@ -41,20 +41,33 @@ type dashMeta struct {
 func (s *StandardSearchService) DoDashboardQuery(ctx context.Context, user *backend.User, orgId int64, query DashboardQuery) *backend.DataResponse {
 	rsp := &backend.DataResponse{}
 
-	if user == nil || user.Role != string(models.ROLE_ADMIN) {
```
[the remark this answers is not in the document]
Comment (FZambia) [reply pkg/services/searchV2/service.go:44]: Mm, I think I only now realized that we aim to have search panel? Is this correct? Previously I thought this all only to quickly test out search results.

File: pkg/services/searchV2/auth.go [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
 
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/sqlstore/permissions"
 )
 
 // ResourceFilter checks if a given a uid (resource identifier) check if we have the requested permission
-type ResourceFilter func(uid string) bool
+type ResourceFilter interface {
+	check(uid string) bool
```
Comment (ryantxu) [inline pkg/services/searchV2/auth.go:13]: any reason this is better than having a function?

It does let us put the lookup map in a struct... but I think we could do the same in a closure

File: pkg/services/searchV2/auth.go [hunk 5]
```diff
@@ -19,20 +22,62 @@ type simpleSQLAuthService struct {
...
+		OrgId:           user.OrgId,
+		Dialect:         a.sql.Dialect,
+		UserId:          user.UserId,
+		PermissionLevel: models.PERMISSION_VIEW,
 	}
 
-	// TODO: find all matching IDs for the user and check results
-	fmt.Printf("TODO, SELECT all ids for user: %v\n", a.sql)
-	return alwaysFalseFilter, nil
+	rows := make([]*dashIdQueryResult, 0)
```
Comment (ryantxu) [inline pkg/services/searchV2/auth.go:41]: better as a `map[string]bool` -- since we will support random lookup by uid

**Pull request merged**